### PR TITLE
Write network configuration to have wicked understand ovs (bsc#930986)

### PR DESCRIPTION
**Pull request by vuntz**
This is based on https://en.opensuse.org/Portal:Wicked/OpenvSwitch and
comment 9 in the bug.

https://bugzilla.suse.com/show_bug.cgi?id=930986

Depends on https://github.com/crowbar/barclamp-network/pull/327

### CODE REVIEW

**Comment by vuntz** [comment]
cc @rhafer since he might have an opinion on the uglyness of the approach ;-)

File: chef/cookbooks/neutron/recipes/common_agent.rb [hunk 1]
```diff
@@ -157,53 +162,140 @@
...
+
+    if node[:platform] == "suse" && node[:platform_version].to_f >= 12.0
+      # on SLE12, we create proper network configuration for OVS bridges, see
+      # https://en.opensuse.org/Portal:Wicked/OpenvSwitch
+
+      # We need to create a static ifcfg file so that wicked will know that the
+      # ovs bridge usurps the interface address. This also means that we need
+      # to override the variables we pass to the template for the ifcfg of the
+      # interface.
+      bound_if_resource = resources(:template => "/etc/sysconfig/network/ifcfg-#{bound_if}")
```
Comment (aplanas) [inline chef/cookbooks/neutron/recipes/common_agent.rb]: Isn't the creation and management of bridges and interfaces a responsibility of barclamp-network?

I was testing that if I check the bridge option for in interface in the node information, and I include the recipe from network, this will update (wrongly) the interfaces.

IIUC one of the risks of managing the interfaces here is that barclamp-network can undo this.

Comment (vuntz) [inline chef/cookbooks/neutron/recipes/common_agent.rb]: > Isn't the creation and management of bridges and interfaces a responsibility of barclamp-network?

Yes and no. The issue is that the network cookbook doesn't have information about what is needed here, and that's information internal to the neutron cookbook.

> I was testing that if I check the bridge option for in interface in the node information, and I include the recipe from network, this will update (wrongly) the interfaces.

Hrm, not sure I understand. Can you rephrase or give more details?

> IIUC one of the risks of managing the interfaces here is that barclamp-network can undo this.

When would this happen? As long as the neutron cookbook is being used, then it will not be undone.

**Comment by aplanas** [comment]
+1 Tested with wicked updated

File: chef/cookbooks/neutron/recipes/common_agent.rb (hunk 1 above)
Comment (dirkmueller) [inline chef/cookbooks/neutron/recipes/common_agent.rb]: I wonder if this could be moved to barclamp-network instead. it shouldn't be impossible to figure out that we potentially run neutron on that node.. just need to search for an active proposal and then deploy the usurp part there..

**Comment by rsalevsky** [comment]
@vuntz, this repository got merged into [crowbar-openstack](https://github.com/crowbar/crowbar-openstack).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by rsalevsky** [comment]
Moved to https://github.com/crowbar/barclamp-neutron/pull/245
